### PR TITLE
8203925: tools/javac/importscope/T8193717.java ran out of java heap

### PR DESCRIPTION
**Pull request by lgxbslgx**
Hi all,

Because [JDK-8225054-PATCH](https://github.com/openjdk/jdk/commit/827e5e32264666639d36990edd5e7d0b7e7c78a9) modified `T8193717.java`, changing `private static final int CLASSES = 500000;` to `private static final int CLASSES = 50000;`(500K -> 50K). The error, `out of java heap`, no longer occurs now. And the pre-submit tests of this patch passed. I think `tools/javac/importscope/T8193717.java` could be removed from `ProblemList.txt`.

Thank you for taking the time to review.

Best Regards.

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed

### Issue
 * [JDK-8203925](https://bugs.openjdk.java.net/browse/JDK-8203925): tools/javac/importscope/T8193717.java ran out of java heap

### Reviewers
 * [Vicente Romero](https://openjdk.java.net/census#vromero) (@vicente-romero-oracle - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.java.net/jdk pull/1998/head:pull/1998` \
`$ git checkout pull/1998`

Update a local copy of the PR: \
`$ git checkout pull/1998` \
`$ git pull https://git.openjdk.java.net/jdk pull/1998/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 1998`

View PR using the GUI difftool: \
`$ git pr show -t 1998`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.java.net/jdk/pull/1998.diff">https://git.openjdk.java.net/jdk/pull/1998.diff</a>

</details>

### CODE REVIEW

File: test/langtools/ProblemList.txt [hunk 1]
```diff
@@ -49,7 +49,6 @@ tools/javac/annotations/typeAnnotations/referenceinfos/Lambda.java
 tools/javac/annotations/typeAnnotations/referenceinfos/NestedTypes.java         8057687    generic-all    emit correct byte code an attributes for type annotations
 tools/javac/warnings/suppress/TypeAnnotations.java                              8057683    generic-all    improve ordering of errors with type annotations
 tools/javac/modules/SourceInSymlinkTest.java                                    8180263    windows-all    fails when run on a subst drive
-tools/javac/importscope/T8193717.java                                           8203925    generic-all    the test requires too much memory
```
Comment (vicente-romero-oracle) [inline test/langtools/ProblemList.txt:52]: I believe that we want to do some more research regarding why the test fails for a large number of imports

**Comment by lgxbslgx** [comment]
@vicente-romero-oracle Thank you for your comment. When the number of the `import` statements increases, the `JCImport` tree and other objects in `javac` would increase, too.(In other words, the memory would be gradually used, too) There is no doubt that `out of heap` would occur if the number of the `import` statements becomes so large.

---
I read the original issue [JDK-8193717](https://bugs.openjdk.java.net/browse/JDK-8193717) and found the original number should be 50k instead of 500k. The quote is shown below. We can consider that the 500k is the wrong value.
> Compiling a generated file with about 50 thousand imports

And then, [JDK-8225054-PATCH](https://github.com/openjdk/jdk/commit/827e5e32264666639d36990edd5e7d0b7e7c78a9) fixed it(500k -> 50k). The [JDK-8225054-PATCH](https://github.com/openjdk/jdk/commit/827e5e32264666639d36990edd5e7d0b7e7c78a9) is a big feature(Compiler implementation for records) and some details may be omitted. I think that the authors of the patch forgot to remove `tools/javac/importscope/T8193717.java` from the `ProblemList` when they fixed the bug(500k -> 50k).

---
I merged the master branch code just now to restart the `Pre-submit test`. If the test `tools/javac/importscope/T8193717.java` passes at the `Pre-submit test`, I think it is good to remove it from the `ProblemList`.

What's your opinion? Maybe the authors of the [JDK-8225054-PATCH](https://github.com/openjdk/jdk/commit/827e5e32264666639d36990edd5e7d0b7e7c78a9) could give more information. Any idea is appreciated.

**Review by vicente-romero-oracle — APPROVED** [review]

**Pull request closed without merging**